awx/sso/backends.py

```diff
@@ -383,31 +381,73 @@ def on_populate_user(sender, **kwargs):
             force_user_update = True
             logger.warning('LDAP user {} has {} > max {} characters'.format(user.username, field, max_len))
 
-    # Update organization membership based on group memberships.
     org_map = getattr(backend.settings, 'ORGANIZATION_MAP', {})
+    team_map = getattr(backend.settings, 'TEAM_MAP', {})
+
+    # Move this junk into save of the settings for performance later, there is no need to do that here
+    #    with maybe the exception of someone defining this in settings before the server is started?
+    # ==============================================================================================================
+
+    # Get all of the names of orgs in the DB and create any new org defined in LDAP that does not exist in the DB
+    existing_org_names = list(Organization.objects.all().values_list('name', flat=True))
+    for org_name in org_map.keys():
+        if org_name not in existing_org_names:
+            Organization.objects.get_or_create(name=org_name)
```

Comment:
If the org name isn't in `existing_org_names`, then deterministically, you create, right? Then I would opt for `create` instead of `get_or_create`, as it will be more performant and clear. If you are concerned about race conditions, this isn't the way to do things, you should do a try-except if you are really worried about it being created while in the loop so that you can later on make correct decisions.